### PR TITLE
Add a mutator initializing ForceFree::Tags::NsInteriorMask

### PR DESCRIPTION
**Pull request by yoonso0-0**
## Proposed changes

This is a follow up of #5232 , a mutator assigning the `std::optional` masking tag in the initialization phase for NS magnetosphere initial data.

Apart from the last one, other commits are from : 

- [ ] #5281 

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp
Comment (yoonso0-0) [inline src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp]: `ImposeMhdConditionInsideNS` is the name of another mutator for which I'll submit another PR soon.

File: src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp
Comment (yoonso0-0) [inline src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp]: @wthrowe This is the actual usage of #5272 , I welcome any better idea for this.

Comment (wthrowe) [reply src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp]: How about adding
```c++
  static auto variables(const tnsr::I<DataVector, 3>& coords,
                        tmpl::list<Tags::NsInteriorMask> /*meta*/)
      -> tuples::TaggedTuple<Tags::NsInteriorMask>;
```
or some similar method to the initial data classes?

Comment (yoonso0-0) [reply src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp]: and change this mutator to just call the `variables` function?

Comment (wthrowe) [reply src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp]: Right.

Comment (yoonso0-0) [reply src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp]: Sounds good.

File: src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp
Comment (wthrowe) [inline src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp]: Use the inertial frame because that's how the fields are defined.  The result can't vary from call to call because initialization mutators are only called once.

File: src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp
Comment (wthrowe) [inline src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp]: Do the call unconditionally.  I think if you use `Tags::AnalyticSolutionOrData` in `argument_tags` you can even avoid the `call_with_dynamic_type`.  The function in `src/Evolution/Initialization/InitialData.hpp` might be useful, but I'm not sure on that one.

File: src/PointwiseFunctions/AnalyticData/ForceFree/RotatingDipole.cpp
Comment (wthrowe) [inline src/PointwiseFunctions/AnalyticData/ForceFree/RotatingDipole.cpp]: Is this true?  I don't see anything in the fields doing this, or is the NS region arbitrary?

**Review by wthrowe — CHANGES_REQUESTED** [review]
Haven't reviewed the test.  Will do that once we have the design ironed out.

File: src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp
Comment (yoonso0-0) [reply src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp]: Hmm maybe I'll remove this `\note` comment and use inertial frame coordinates for initialization.

File: src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp
Comment (yoonso0-0) [reply src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp]: 1. `Forcefree` system is using runtime initial data based on the factory infrastructure, which was why I used `call_with_dynamic_type`. Not sure if there is a way around.
2. If run unconditionally, wouldn't this raise "tag is not retrievable" error for other types of initial data? e.g. simple 1D wave tests which we don't need masking. My idea was to have the 
```
  static auto variables(const tnsr::I<DataVector, 3>& coords,
                        tmpl::list<Tags::NsInteriorMask> /*meta*/)
      -> tuples::TaggedTuple<Tags::NsInteriorMask>;
```
only declared for NS magnetosphere initial data.

File: src/PointwiseFunctions/AnalyticData/ForceFree/RotatingDipole.cpp
Comment (yoonso0-0) [reply src/PointwiseFunctions/AnalyticData/ForceFree/RotatingDipole.cpp]: We set `c=1` in this evolution system so the length (=time) scale can be arbitrarily rescaled. I could add some option like `NsRadius`, but thought it'd be convenient to just fix `R_* = 1.0` and interpret all the results from this initial data in terms of `R_S`. For example, `t=1.0` in the code unit would correspond to `R_S / c` in physical unit.

File: src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp
Comment (wthrowe) [reply src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp]: 1. Yes, you're right.  I was looking at code that hasn't been converted to run-time yet.
2. You'll have to either implement it for the others, or make it it's own function instead of a `variables` specialization and check if that function is callable.

File: src/PointwiseFunctions/AnalyticData/ForceFree/RotatingDipole.cpp
Comment (wthrowe) [reply src/PointwiseFunctions/AnalyticData/ForceFree/RotatingDipole.cpp]: No that's fine.  I was expecting the fields to have a different form inside the star, but I see now that that isn't the case.

File: src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp
Comment (yoonso0-0) [reply src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp]: Okay, I'll change this to something other than the `variables` specialization.

**Comment by yoonso0-0** [comment]
I pushed the fixed version, and since there were multiple fixes I just squashed into one. The mutator itself actually became simpler.

File: src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp
Comment (wthrowe) [reply src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp]: Use `CREATE_IS_CALLABLE` instead of hardcoding the solutions that need masks.

File: src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp
Comment (wthrowe) [inline src/Evolution/Systems/ForceFree/MaskNeutronStarInterior.hpp]: Move this information to the solution, since that's where it is now implemented.  (Or just remove it if you think it's covered sufficiently there already.)

**Review by wthrowe — CHANGES_REQUESTED** [review]

**Comment by yoonso0-0** [comment]
Okay, I could remove the `enum` class thing!

**Comment by wthrowe** [comment]
This is ready to be squashed, but I suggest waiting until #5281 is merged and you can rebase on it to reduce the stress on CI a bit.

**Comment by yoonso0-0** [comment]
Okay, thanks for the review! I'll close #5272

**Review by wthrowe — APPROVED** [review]
@yoonso0-0 This is ready to go, yes?  I think the dependent label is just left over from before the last update.

**Comment by yoonso0-0** [comment]
@wthrowe Yes

**Comment by yoonso0-0** [comment]
Rebased on the latest develop after a couple of CI failures

**Pull request merged**
